sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py

```diff
@@ -55,6 +55,9 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         if not scopes:
             raise ValueError("'get_token' requires at least one scope")
 
+        if not self._initialized:
+            self._initialize()
```

Comment:
You have 
def _initialize(self):
        if self._initialized:
            return
seems like you dup the work

Reply:
It's defense against someone calling `_initialize` twice. But that shouldn't happen and if it does, the consequence is recreating the client, which isn't destructive. I removed the double-check.

Reply:
To make sure we are on same page, what I meant was:

Keep 
```python
def _initialize(self):
        if self._initialized:
            return
```
But modify
```python
if not self._initialized:
            self._initialize()
```
to
```python
self._initialize()
```

Reply:
Ah, I see. I implemented it this way to avoid an additional call.

Reply:
If I understand correctly, calling into _initialize() twice may cause problem. 

You have allow_unencrypted = self._client_kwargs.pop("allow_unencrypted_cache", False) in your code which makes second call may have different value.

I agree with your original statement:
It's defense against someone calling _initialize twice.

Reply:
Good eye, and you've reignited my paranoia :suspect: